--- same-C-run-1
+++ same-C-run-2
@@ -1,7 +1,7 @@
 Q3 1.32e linux-x86_64 Sep 13 2026
 ----- FS_Startup -----
 ...found 9 cached paks
-...loaded in 0 milliseconds
+...loaded in 1 milliseconds
 ----------------------
 4073 files in 9 pk3 files
 execing default.cfg
@@ -81,12 +81,11 @@
 broadcast: print "Major^7 connected\n"
 broadcast: print "Major^7 entered the game\n"
 ClientBegin: 1
+Item: 0 weapon_rocketlauncher
 Item: 1 ammo_rockets
-Item: 0 item_armor_body
-Item: 1 weapon_rocketlauncher
-Item: 0 ammo_rockets
-Item: 0 weapon_rocketlauncher
-Item: 1 ammo_bullets
+Item: 1 item_armor_body
+Item: 1 item_health_large
+Item: 1 item_health_large
 ----- Server Shutdown (Server quit) -----
 ==== ShutdownGame ====
 ShutdownGame:
